### ISSUE TITLE
bug: Kung Fu false positive

### ISSUE THREAD

**Issue by krasnoperov** [jo3-l/obscenity#67]
### Expected behavior

matcher.hasMatch('Kung-Fu') returns true

### Actual behavior

matcher.hasMatch('Kung-Fu') returns false

### Minimal reproducible example

```typescript
import assert from 'node:assert'

import {
  englishDataset,
  englishRecommendedTransformers,
  RegExpMatcher,
} from 'obscenity'

const matcher = new RegExpMatcher({
  ...englishDataset.build(),
  ...englishRecommendedTransformers,
})

assert.equal(matcher.hasMatch('Kung-Fu'), false)
assert.equal(matcher.hasMatch('Kung Fu'), false)
assert.equal(matcher.hasMatch('Kung Fu Panda'), false)

// This one actually works
assert.equal(matcher.hasMatch('KungFu'), false)
```

### Steps to reproduce

1. Run the code above
2. It falls with assert exception

### Node.js version

v20.15.0

### Obscenity version

0.2.1

### Priority

- [X] Low
- [ ] Medium
- [ ] High

### Terms

- [X] I agree to follow the project's Code of Conduct.
- [X] I have searched existing issues for similar reports.

**Comment by jo3-l** [comment]
The default dataset contains the pattern `|fu|`, which (correctly, but undesirably) matches on the `-Fu` in `Kung-Fu`. There are two potential ways we could fix this issue:
1. Remove the `|fu|` pattern entirely, or
2. Whitelist `Kung-Fu` and leave the `|fu|` pattern untouched.

I am leaning toward 2) at the moment: the `|fu|` pattern seems useful in general, and I cannot think of any other egregious false positives other than the instance you report. What do you think?

**Comment by krasnoperov** [comment]
I think that whitelisting Kung-Fu is a good option here. Also, it is possible to handle any future false positives by adding them to the whitelist as they arise.

**Comment by jo3-l** [comment]
I released v0.3.0 with the fix (please ignore v0.2.2; main and dev got out of sync and some nasty conflicts resulted.) Thanks again for the report.
